mithril-client-wasm/src/certificate_verification_cache.rs

```diff
@@ -0,0 +1,441 @@
+            let now = Utc::now();
+            let cache = LocalStorageCertificateVerifierCache::new(
+                "store_in_empty_cache_add_new_item_that_expire_after_parametrized_delay",
+                expiration_delay,
+            );
+            cache
+                .store_validated_certificate("hash", "parent")
+                .await
+                .unwrap();
+
+            let cached = cache
+                .get_cached_value("hash")
+                .expect("Cache should have been populated");
+
+            assert_eq!(1, cache.len());
+            assert_eq!("parent", cached.previous_hash);
+            assert!(cached.expire_at - now >= expiration_delay);
+        }
+
+        #[wasm_bindgen_test]
+        async fn store_new_hash_push_new_key_at_end_and_dont_alter_existing_values() {
+            let cache = LocalStorageCertificateVerifierCache::new(
+                "store_new_hash_push_new_key_at_end_and_dont_alter_existing_values",
+                TimeDelta::hours(1),
+            )
+            .with_items([
+                ("existing_hash", "existing_parent"),
+                ("another_hash", "another_parent"),
+            ]);
+            cache
+                .store_validated_certificate("new_hash", "new_parent")
+                .await
+                .unwrap();
+
+            assert_eq!(
+                HashMap::from_iter([
+                    ("existing_hash".to_string(), "existing_parent".to_string()),
+                    ("another_hash".to_string(), "another_parent".to_string()),
+                    ("new_hash".to_string(), "new_parent".to_string()),
+                ]),
+                cache.content()
+            );
+        }
+
+        #[wasm_bindgen_test]
+        async fn storing_same_hash_update_parent_hash_and_expiration_time() {
+            let expiration_delay = TimeDelta::days(2);
+            let now = Utc::now();
+            let cache = LocalStorageCertificateVerifierCache::new(
+                "storing_same_hash_update_parent_hash_and_expiration_time",
+                expiration_delay,
+            )
+            .with_items([("hash", "first_parent"), ("another_hash", "another_parent")]);
+
+            cache
+                .store_validated_certificate("hash", "updated_parent")
+                .await
+                .unwrap();
+
+            let updated_value = cache
+                .get_cached_value("hash")
+                .expect("Cache should have been populated");
+
+            assert_eq!(2, cache.len());
+            assert_eq!(
+                Some("another_parent".to_string()),
+                cache.get_previous_hash("another_hash").await.unwrap(),
+                "Existing but not updated value should not have been altered, content: {:#?}, now: {:?}",
+                local_storage_content(), now,
+            );
+            assert_eq!("updated_parent", updated_value.previous_hash);
+            assert!(updated_value.expire_at - now >= expiration_delay);
```

Comment:
I'm not sure that we prove that the `expire_at` was updated here. 
We may check that the value has change.